src/main/java/nextstep/ladder/domain/Line.java

```diff
@@ -0,0 +1,55 @@
+package nextstep.ladder.domain;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class Line {
+    private List<Point> points;
+
+    public Line(List<Point> points) {
+        this.points = points;
+        checkOverLabLine();
+    }
+
+    public Line(int countOfPerson) {
+        this.points = new ArrayList<>(countOfPerson);
+
+        points.add(new Point(RandomLine::getRandomValue));
+
+        for (int i = 1; i < countOfPerson - 1; i++) {
+            int prevIndex = i - 1;
+            points.add(new Point(() -> isBeforeDrawLine(prevIndex)));
+        }
+
+        checkOverLabLine();
+    }
+
+    private boolean isBeforeDrawLine(int prevIndex) {
+        return !findPoint(prevIndex).isActive() && RandomLine.getRandomValue();
+    }
+
+    private Point findPoint(int index) {
+        return points.get(index);
+    }
+
+    private void checkOverLabLine() {
+        boolean foundOverLab = IntStream.range(0, points.size() - 1)
+                .anyMatch(this::isSamePosition);
+
+        if (foundOverLab)
```

Comment:
```suggestion
        if (isFoundOverLab())
```

복잡한 스트림 구조를 의미있는 이름과 함께 `단일책임`,`순수함수` 형태로 표현하면 어떨까요?

Reply:
위처럼 변경하면, 해당 함수가 단일책임을 갖는다는 건 이해했습니다!
근데 순수함수 형태로 표현한다는게 잘 이해가 안가는데, 자세히 설명해주실 수 있을까요?